pkg/controller/direct/monitoring/refs.go

```diff
@@ -0,0 +1,105 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package monitoring
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	krm "github.com/GoogleCloudPlatform/k8s-config-connector/apis/monitoring/v1beta1"
+	refs "github.com/GoogleCloudPlatform/k8s-config-connector/apis/refs/v1beta1"
+	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/clients/generated/apis/k8s/v1alpha1"
+	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/controller/direct/references"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func normalizeResourceName(ctx context.Context, reader client.Reader, src client.Object, ref *v1alpha1.ResourceRef) (*v1alpha1.ResourceRef, error) {
```

Comment:
Do you want to move this to `controller/direct/reference/` or something that can be sharable? I feel like this is a good piece to reuse and improve as we building up more resources.

Reply:
Yes, but my "rule of thumb" is to move it when we have two copies.  For this particular one, we only have one copy  (this is actually very specialized - there's a field called "resourceName" in MonitoringDashboard.  Note that this method does call into shared functions for most of its logic!

Reply:
Actually I guess this is resolving a resource of arbitrary kind, so is more reusable than I first thought.  Still, it's our first ... let's move it when we have two or three instances.